src/item/Item.js

```diff
@@ -1587,6 +1587,17 @@ new function() { // Injection scope for various item event handlers
         return this._index;
     },
 
+    setIndex: function(index) {
+        var parent = this._parent;
+        if (!parent) return;
+        var parentChildren = parent._children;
+        if (!parentChildren) return;
+        if (!parentChildren[index]) {
+            index = undefined;
+        }
+        parent.insertChildren(index, [this]);
+    },
```

Comment:
Could be shortened to:

```js
    setIndex: function(index) {
        var parent = this._parent,
            children = parent && parent._children;
        if (children) {
            parent.insertChildren(index in children ? index : undefined, [this]);
        }
    },
```

Reply:
Thanks @lehni for reviewing! I like shorter one, too ;)
I fixed as you proposed.